[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd

voznesenskym opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108991
* #108846

[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

voznesenskym committed:
[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd

[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd

[ghstack-poisoned]

voznesenskym commented on docs/source/logging.rst:
```diff
@@ -65,6 +65,9 @@ Artifacts:
             Whether to emit the joint forward-backward graph generated by AOTAutograd. Default: ``False``
 
+        ``compiled_autograd``
+            Wether to emit logs from compiled_autograd. Defaults: ``False``
```
Whether

voznesenskym committed:
Update on "[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd"


[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

ezyang commented on torch/_logging/_registrations.py:
```diff
@@ -45,6 +45,11 @@
 )
+register_artifact(
+    "compiled_autograd",
+    "Prints various logs in compiled_autograd, including but not limited to the graphs. Useful for debugging compiled_autograd.",
```
Nit: as you picked this as an artifact, it would be more symmetric with the other existing logs to name it `compiled_autograd_graph` and have it only print the graph.

voznesenskym replied:
Good nit. I considered it. I'm gonna use it for more than that, and I always want the graph.

voznesenskym commented:
@pytorchbot merge

voznesenskym commented:
bruh logging tests

voznesenskym committed:
Update on "[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd"


[Easy] log graphs in compiled_autograd if TORCH_LOGS=compiled_autograd

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym commented:
@pytorchbot merge

voznesenskym commented:
@pytorchbot merge -f "CI is flaky"
